include/linux/kfence.h

```diff
@@ -39,8 +39,7 @@ extern struct static_key_false kfence_allocation_key;
  */
 static __always_inline bool is_kfence_address(const void *addr)
 {
-	return unlikely((char *)addr >= __kfence_pool &&
-			(char *)addr < __kfence_pool + KFENCE_POOL_SIZE && addr);
+	return unlikely((unsigned long)((char *)addr - __kfence_pool) < KFENCE_POOL_SIZE && addr);
```

Comment:
Ok, now checking for `addr` makes sense. But maybe it should look like: `likely(addr) && unlikely((unsigned long)((char *)addr - __kfence_pool) < KFENCE_POOL_SIZE)` then?

Reply:
The non-NULL check is still needed in case KFENCE was never initialized.

I think we want to write this so that the non-NULL check is in the unlikely slow-path:
```
Also, you could do the NULL *after* the range check said "true". That
way the NULL check would be on the slowpath and have basically no
performance impact.
```

Reply:
Also, hopefully by only having 1 branch for the range check, we train the CPU's branch predictor enough so that it'll predict the range check as false and we do not even notice the pointer load.

Reply:
Added a comment.

Reply:
Ok, makes sense.